authservice/src/main/java/auth/proximity/authservice/controller/AuthController.java

```diff
@@ -174,7 +178,11 @@ public ResponseEntity<ResponseDto> updatePassword(@RequestParam String email, @R
         return ResponseEntity.ok(new ResponseDto("200", "Password updated successfully"));
     }
     @PutMapping("/public/update-profile-picture")
-    public ResponseEntity<String> uploadProfilePicture(@RequestParam String email, @ModelAttribute ProfilePictureUpdateRequest profilePictureUpdateRequest) {
+    public ResponseEntity<String> uploadProfilePicture(@AuthenticationPrincipal UserDetailsImpl userDetails, @ModelAttribute ProfilePictureUpdateRequest profilePictureUpdateRequest) {
+        if (userDetails == null) {
```

Comment:
This is redundant. Use role base to restrict the endpoint and in that case you wouldn't need to check if the principal is null.

If the principal is null, the authentication/authorization filter will bounce it and it won't even get to ur controller.